docs/docs/dev_docs/tutorials/token_portal/withdrawing_from_l1.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Withdrawing from L1
+---
+
+This is where we have tokens on Aztec and want to withdraw them back to L1 (i.e. burn them on L2 and mint on L1). Withdrawing from L1 will be public.
+
+## Withdrawing publicly
+
+Go back to your `main.nr` and paste this:
+
+#include_code exit_to_l1_public /yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/main.nr rust
+
+For this to work we will need this helper function:
+
+#include_code get_withdraw_content_hash /yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/util.nr rust
+
+**What’s happening here?**
+
+The `exit_to_l1_public` function enables anyone to withdraw their L2 tokens back to L1 publicly. This is done by burning tokens on L2 and then creating an l2->l1 message.
+
+1. Like with our deposit function, we need to create the l2 to l1 message. The content is the _amount_ to burn, the recipient address, and who can execute the withdraw on the L1 portal on behalf of the user. It can be `0x0` for anyone, or a specified address.
```

Comment:
```suggestion
1. Like with our deposit function, we need to create the L2 to L1 message. The content is the _amount_ to burn, the recipient address, and who can execute the withdraw on the L1 portal on behalf of the user. It can be `0x0` for anyone, or a specified address.
```